install/swiftly-install.sh

```diff
@@ -128,6 +128,18 @@ install_system_deps () {
         return
     fi
 
+    if [[ "$docker_platform_name" == "debian" ]]; then
```

Comment:
FYI I filed https://github.com/swiftlang/swift-docker/issues/409 to cover adding the proper Debian 12 images so we don't need the hard code or fallback to swift-ci's list.

Reply:
I think hard coding the dependencies is a much better option, especially for all the ubuntu and debian platforms. Trying to parse multiple dockerfiles to obtain the same dependency list is over engineering what should be a very simple task.
The following list of dependencies covers all current versions of ubuntu and debian with swift releases.
```bash
build-essential 
git 
gnupg2  
libcurl4 
libedit2 
libncurses-dev 
libpython3-dev  
libxml2  
libz3-dev 
pkg-config 
python3 
python3-lldb 
tzdata 
unzip 
zlib1g-dev
```
I'm sure a similar simplified list could be made for RHEL platforms.

If in some future swift release some extra dependencies are required, then it is a very quick and simple task to add it to the list.

Simple to update, simple to maintain and guaranteed to work correctly.